Mancala.hs

```diff
@@ -11,18 +11,18 @@ type Board = (Side, Side)  -- TODO: not sure if using just a list is any better
 type InternalState = Board 
 data Action = Action Int -- index of a selected pocket (non-empty)
          deriving (Ord,Eq)
-data State = State InternalState ([Action], [Action])  
+data State = State InternalState ([Action], [Action])
          deriving (Ord, Eq, Show)
 
 data Result = EndOfGame Double State    -- end of game: value, starting state
             | EndOfTurn State        -- the other player continues 
-            | ContinueTurn State        -- current player takes another turn
+            | ContinueTurn State        -- current player takes another turn -- might not be necessary. can just call person_play or opponent_play again
          deriving (Eq, Show)
 type Game = Action -> State -> Result
 type Player = State -> Action
 
 side_start = [4, 4, 4, 4, 4, 4, 0]
-actions_start = [Action i | i <- [0..5]] -- only not-empty pockets are valid actions 
+actions_start = [Action i | i <- [0..5]] -- only not-empty pockets are valid actions -> Guarding for in play (probably not the best idea) 
```

Comment:
I was thinking as we update the board we update the available actions (non-empty pockets on each side). So we only provide those options to the player in the prompt. And in the compuer_play/person_play we can just check what the player types in against the available actions.

Reply:
In that case, we just need a function to parse for available states then. I can remove the guard from Play.hs.